src/extension/background-script/actions/ln/makeInvoice.js

```diff
@@ -1,14 +1,29 @@
 import PubSub from "pubsub-js";
 import state from "../../state";
+import utils from "../../../../common/lib/utils";
 
 const makeInvoice = async (message, sender) => {
   if (message.args.memo === undefined) {
     message.args.memo = "Alby invoice memo";
   }
   PubSub.publish(`ln.makeInvoice.start`, message);
 
-  const amount = parseInt(message.args.amount);
+  let amount;
   const memo = message.args.memo;
+  if (message.args.amount) {
+    amount = parseInt(message.args.amount);
+  } else {
+    const response = await utils.openPrompt({
```

Comment:
I think we should refactor this pattern as it had been problematic in the past (e.g. impossible to have a loading indicator) 

I think we should do that similarly to what we are doing here: https://github.com/getAlby/lightning-browser-extension/blob/master/src/app/screens/ConfirmPayment/index.jsx#L22

Reply:
Hmm the https://webln.dev/#/api/make-invoice fails if the makeInvoice is not awaiting a response. Not sure how to integrate this pattern here.

Reply:
we have to have the response, but here we did it like this: 
prompt -> does another action call and then returns that response back

https://github.com/getAlby/lightning-browser-extension/blob/master/src/app/screens/ConfirmPayment/index.jsx#L29-L34

Reply:
Not sure why I didn't got this working yesterday, but working now :)